src/System.Private.ServiceModel/src/Internals/System/Runtime/TimeoutHelper.cs

```diff
@@ -294,12 +280,20 @@ public static Task<CancellationToken> FromTimeoutAsync(int millisecondsTimeout)
                 {
                     // Since this thread was successful reserving a spot in the cache, it would be the only thread
                     // that construct the CancellationTokenSource
-                    var token = new CancellationTokenSource((int)(targetTime - currentTime)).Token;
+                    var tokenSource = new CancellationTokenSource((int)(targetTime - currentTime));
+                    var token = tokenSource.Token;
 
                     // Clean up cache when Token is canceled
                     token.Register(t => {
                         Task<CancellationToken> ignored;
```

Comment:
It might be work make this delegate a static to avoid accidental closure allocation.